packages/flutter_tools/lib/src/mdns_discovery.dart

```diff
@@ -229,10 +229,20 @@ class MDnsVmServiceDiscovery {
       final Set<String> uniqueDomainNamesInResults = <String>{};
 
       // Listen for mDNS connections until timeout.
-      final Stream<PtrResourceRecord> ptrResourceStream = client.lookup<PtrResourceRecord>(
-        ResourceRecordQuery.serverPointer(dartVmServiceName),
-        timeout: timeout
-      );
+      final Stream<PtrResourceRecord> ptrResourceStream;
+
+      try {
+        ptrResourceStream = client.lookup<PtrResourceRecord>(
+            ResourceRecordQuery.serverPointer(dartVmServiceName),
+            timeout: timeout
+        );
+      } on SocketException {
+        throwToolExit('You may be having a permissions issue with your IDE. '
```

Comment:
Some folks in the thread reported that updating permissions didn't fix the issue ([example](https://github.com/flutter/flutter/issues/150131#issuecomment-2422824717)).

Should we include the original exception message of `SocketException: Send failed, OS Error: No route to host, errno = 65`? This might be useful information for folks where fixing the permissions doesn't help.

Reply:
I agree, we can print the error and let it throw so we can continue to get crash numbers on this.

Reply:
done - surfaced error and stack trace